src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/Logout.razor

```diff
@@ -1,94 +1,78 @@
+        @if (IsLoading)
+        {
+            <div class="logout-spinner">
+                <i class="fas fa-spinner fa-spin fa-3x"></i>
+            </div>
+            <h2 class="logout-title">Signing out...</h2>
+            <p class="logout-message">Please wait while we sign you out.</p>
+        }
+        else if (HasError)
+        {
+            <div class="logout-icon logout-error">
+                <i class="fas fa-exclamation-triangle"></i>
+            </div>
+            <h2 class="logout-title">Logout Error</h2>
+            <p class="logout-message">@ErrorMessage</p>
+        }
+        else
+        {
+            <div class="logout-icon logout-success">
+                <i class="fas fa-check-circle"></i>
+            </div>
+            <h2 class="logout-title">Signed out successfully</h2>
+            <p class="logout-message">You have been successfully signed out.</p>
+        }
     </div>
 </div>
 
-<style>
-    .logout-container {
-        min-height: 100vh;
-        display: flex;
-        align-items: center;
-        justify-content: center;
-        background: linear-gradient(135deg, #00879c 0%, #006b7d 100%);
-        padding: 20px;
-    }
-
-    .logout-card {
-        background: white;
-        border-radius: 12px;
-        box-shadow: 0 20px 60px rgba(0, 0, 0, 0.1);
-        padding: 40px;
-        text-align: center;
-        max-width: 400px;
-        width: 100%;
-    }
-
-    .logout-icon {
-        color: #00879c;
-        font-size: 4rem;
-        margin-bottom: 20px;
-    }
-
-    .logout-title {
-        color: #495057;
-        font-size: 1.5rem;
-        font-weight: 600;
-        margin-bottom: 15px;
-    }
-
-    .logout-message {
-        color: #6c757d;
-        font-size: 1rem;
-    }
-</style>
-
 @code {
+    [SupplyParameterFromQuery]
+    private string? ReturnUrl { get; set; }
+
     protected override async Task OnInitializedAsync()
     {
-        try
+        await ExecuteAsync(async () =>
         {
-            Logger.LogInformation("User logout initiated");
+            var command = new LogoutCommand
+            {
+                ReturnUrl = ReturnUrl
+            };
+
+            var result = await Mediator.Send(command);
 
-            // Clear session storage first
-            try
+            if (result.IsSuccess && result.Content.HasValue && result.Content.Value.IsSuccess)
             {
-                await ProtectedSessionStore.DeleteAsync("UserSession");
-                await ProtectedSessionStore.DeleteAsync("TenantContext");
+                // Wait a moment to show the success message
+                await Task.Delay(1500);
+                
+                // Navigate to the redirect URL or home page
+                var redirectUrl = result.Content.Value.RedirectUrl ?? "/";
+                Navigation.NavigateTo(redirectUrl, true);
             }
-            catch (Exception ex)
+            else if (result.Content.HasValue)
```

Comment:
On Failure responses (e.g., network/service errors), result.Content may be empty. Show result.Error to avoid falling back to a generic message.

```
              else if (result.IsFailure)
+            {
+                // Transport/technical failure
+                SetError(result.Error ?? "Logout failed. Please try again.");
+            }
+            else if (result.Content.HasValue)
             {
                 // Logout failed with a specific error message
                 SetError(result.Content.Value.ErrorMessage ?? "Logout failed. Please try again.");
             }
```